services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.controller.ts

```diff
@@ -1,116 +1,324 @@
-  // @ApiResponse({
-  //   status: HttpStatus.OK,
-  //   description: 'Return programFspConfigurationEntity by program id.',
-  // })
-  // @Get(':programId/fsp-configuration')
-  // public async findByProgramId(
-  //   @Param('programId', ParseIntPipe)
-  //   programId: number,
-  // ): Promise<ProgramFinancialServiceProviderConfigurationEntity[]> {
-  //   return this.programFspConfigurationService.findByProgramId(programId);
-  // }
+  @AuthenticatedUser({ isAdmin: true })
+  @ApiOperation({
+    summary:
+      'Create a Financial Service Provider Configuration for a Program. Financial Service Provider Configuration properties',
+  })
+  @ApiParam({ name: 'programId', required: true, type: 'integer' })
+  @ApiResponse({
+    status: HttpStatus.CREATED,
+    description:
+      'The Financial Service Provider Configuration has been successfully created.',
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description: 'Program does not exist',
+  })
+  @ApiResponse({
+    status: HttpStatus.BAD_REQUEST,
+    description: 'Bad request. Body or params are malformed',
+  })
+  @ApiResponse({
+    status: HttpStatus.CONFLICT,
+    description:
+      'Program Financial Service Provider Configuration with same name already exists',
+  })
+  @Post(':programId/financial-service-provider-configurations')
+  public async create(
+    @Body()
+    programFspConfigurationData: CreateProgramFinancialServiceProviderConfigurationDto,
+    @Param('programId', ParseIntPipe)
+    programId: number,
+  ): Promise<ProgramFinancialServiceProviderConfigurationResponseDto> {
+    return await this.programFinancialServiceProviderConfigurationsService.validateAndCreate(
+      programId,
+      programFspConfigurationData,
+    );
+  }
 
-  // @AuthenticatedUser({ isAdmin: true })
-  // @ApiOperation({
-  //   summary: 'Create ProgramFspConfigurationEntity for a program',
-  // })
-  // @ApiParam({ name: 'programId', required: true, type: 'integer' })
-  // @ApiResponse({
-  //   status: 201,
-  //   description:
-  //     'The programFspConfigurationEntity has been successfully created.',
-  // })
-  // @ApiResponse({ status: HttpStatus.BAD_REQUEST, description: 'Bad request.' })
-  // @ApiResponse({ status: HttpStatus.FORBIDDEN, description: 'Forbidden.' })
-  // @Post(':programId/fsp-configuration')
-  // public async create(
-  //   @Body() programFspConfigurationData: CreateProgramFspConfigurationDto,
-  //   @Param('programId', ParseIntPipe)
-  //   programId: number,
-  // ): Promise<number> {
-  //   return await this.programFspConfigurationService.create(
-  //     programId,
-  //     programFspConfigurationData,
-  //   );
-  // }
+  @AuthenticatedUser({ isAdmin: true })
+  @ApiOperation({
+    summary:
+      'Update a Financial Service Provider Configuration for a Program. Can only update the label and properties. Posting an empty array of properties will delete all properties.',
```

Comment:
Extend description to make clear properties can also/better be updated via PATCH /property. (+ also add/edit description there?)